ast/misc.go

```diff
@@ -1987,6 +1988,21 @@ func (n *GrantStmt) Restore(ctx *RestoreCtx) error {
 			return errors.Annotatef(err, "An error occurred while restore GrantStmt.Users[%d]", i)
 		}
 	}
+	if n.TLSOptions != nil {
+		tlsOptionLen := len(n.TLSOptions)
+		if tlsOptionLen != 0 {
+			ctx.WriteKeyWord(" REQUIRE ")
+		}
+		// Restore `TLSOptions` reversely to keep order the same with original sql
```

Comment:
To avoid this unnatural style, we can define the rule `RequireList` with `RequireList RequireListElement` instead of `RequireListElement RequireList`.